lib/MusicBrainz/Server/Data/Event.pm

```diff
@@ -375,6 +378,35 @@ sub find_by_place
     $self->query_to_list_limited($query, [$place_id], $limit, $offset);
 }
 
+=method find_by_series
+
+Find all events linked to the given series ID via part-of relationships,
+using the provided C<$limit> and C<$offset> parameters for pagination.
+
+=cut
+
+sub find_by_series
+{
+    my ($self, $series_id, $limit, $offset) = @_;
+
+    my $link_type_gid = $PART_OF_SERIES{'event'};
+
+    my $query =
+        'SELECT ' . $self->_columns .'
+           FROM (
+                    SELECT DISTINCT entity0 AS event
+                      FROM l_event_series ar
+                      JOIN link ON ar.link = link.id
+                      JOIN link_type lt ON lt.id = link.link_type
+                     WHERE entity1 = ?
+                       AND lt.gid = ?
+                ) s, ' . $self->_table .'
+          WHERE event.id = s.event
+       ORDER BY event.begin_date_year DESC NULLS LAST, event.begin_date_month DESC NULLS LAST, event.begin_date_day DESC NULLS LAST, event.time DESC NULLS LAST, event.name COLLATE musicbrainz';
```

Comment:
`<<~"SQL"` instead? (You can assign `$self->_columns` and `$self->_table` to variables first to interpolate them.)

For the `ORDER BY`, we can also use `my $order_by = $self->_order_by('date');` like `find_by_events` does.

Reply:
This order by is the opposite direction as the previous one, but it seems fine either way so we might as well be consistent with the newest code.